R/between.R

```diff
@@ -17,12 +17,17 @@ between = function(x, lower, upper, incbounds=TRUE, NAbounds=TRUE) {
   }
   # POSIX check timezone match
   if (is.px(x) && is.px(lower) && is.px(upper)) {
-    tz_match = function(x, y, z) { # NULL match "", else all identical
-      ((is.null(x) || !nzchar(x)) && (is.null(y) || !nzchar(y)) && (is.null(z) || !nzchar(z))) ||
-        (identical(x, y) && identical(x, z))
+    tzs = sapply(list(x,lower,upper), function(x) {
+      tt = attr(x,"tzone",exact=TRUE)
+      if (is.null(tt)) "" else tt
+    })
+    if (tzs[2L]!=tzs[3L]) {
+      stop("'between' lower= and upper= are both POSIXct but have different tzone attributes: ", brackify(tzs[2:3],quote=TRUE), ". Please align their tzone.")
```

Comment:
feels odd to `stop` here but `message` in the `tzone(x)!=tzone(lower)` case

Reply:
It's settled with me after a few hours.  lower and upper are twins really, it would be hard to imagine them originating from different places in the code since they're created together.  Hence leave that as error; do the minimum necessary to pass CoSMoS.    But it's normal for `x` to originate from elsewhere, so treating x vs the twins differently to one twin vs the other twin seems reasonable.